text/0000-offline-buildpackages.md

```diff
@@ -0,0 +1,463 @@
+        },
+        {
+          "sha256": "...",
+          "id": "...",
+          "version": "...",
+          "stacks": [
+            "stack1",
+            "stack2"
+          ]
+        }
+      ]
+    }
+  }
+}
+```
+
+##### Example
+
+```json
+{
+  "buildpacks/nodejs": {
+    "0.0.171": {
+      "api": "0.4",
+      "stacks": [
+        {
+          "id": "io.buildpacks.stacks.bionic"
+        }
+      ],
+      "layerDiffID": "sha256:...",
+      "homepage": "https://buildpacks.io",
+      "assets": [
+        {
+          "sha256": "node12.9.0-sha256",
+          "id": "node",
+          "version": "12.9.0",
+          "name": "Node JS",
+          "uri": "https://asset-location-url",
+          "stacks": [
+            "io.buildpacks.stacks.bionic"
+          ],
+          "metadata": {
+            "license": "MIT"
+          }
+        },
+        {
+          "sha256": "node14.5.0-sha256",
+          "id": "node",
+          "version": "14.5.0",
+          "stacks": [
+            "io.buildpacks.stacks.bionic"
+          ]
+        }
+      ]
+    }
+  }
+}
+```
+
+
+This label let us determine:
+- set of all assets that could be used by a buildpackage
+- set of assets that may be used by an individual buildpack.
+- locations to fetch assets from
+
+### Builder Changes
+
+To keep track of what assets are actually in our builder, and what layer they are contained in we add the label, `io.buildpacks.asset.cache.layers`. Note, this label has already been defined above on asset-cache.
+
+Builders will also require the same changes outlined in the Buildpackage section above to the io.buildpacks.buildpack.layers tag.
+
+
+##### General Format
+```json
+{
+  "buildpacks": [
+    {
+      "id": "buildpack-id",
+      "version": "0.0.1",
+      "homepage": "https://buildpacks.io",
+      "assets": [
+        {
+          "sha256": "asset-1-sha256",
+          "id": "asset-1-id",
+          "version": "1.2.3",
+          "stacks": [
+            "stack1",
+            "..."
+          ]
+        },
+        {
+          "sha256": "asset-2-sha256",
+          "id": "asset-2-id",
+          "version": "2.3.4",
+          "stacks": [
+            "stack1",
+            "..."
+          ]
+        },
+        "..."
+      ]
+    }
+  ]
+}
+```
+
+
+# How it Works
+[how-it-works]: #how-it-works
+
+## Asset cache creation
+
+Asset Cache image creation will be handled by the platform. E.g. `pack package-asset-cache <image-name> <buildpack-uri> --config <optional-path-to-asset-cache.toml>`.
+
+There are two cases depending on weather `<buildpack-uri> points to a buildpackage or a directory containing a `buildpack.toml` file.
+1) Directory containing a `buildpack.toml` file. This method uses the `assets` array to locate download and verify assets.
+2) An existing buildpackage. This will download all assets using `uri`s from the `io.buildpacks.buildpack.layers` label, or re-use assets that are included in the specified buildpackage's layers.
```

Comment:
```suggestion
2) An existing buildpackage. This will download all assets using `uri`s from the `io.buildpacks.buildpack.layers` label.
```
I am assuming this is out of date becaues there is no mechanism for including asset layers in a buildpackage. Is that correct?

Reply:
thanks, missed this edit.